lib/chromedriver.js

```diff
@@ -512,72 +521,47 @@ export class Chromedriver extends events.EventEmitter {
   }
 
   /**
-   * Sync the WebDriver protocol if current on going protocol is W3C or MJSONWP.
-   * Does nothing if this.driverVersion is null.
+   * Determines the driver communication protocol
+   * based on various validation rules.
    *
-   * @returns {typeof PROTOCOLS[keyof typeof PROTOCOLS]}
+   * @returns {keyof PROTOCOLS}
    */
   syncProtocol() {
-    if (!this.driverVersion) {
-      // Keep the default protocol if the driverVersion was unsure.
-      return this.desiredProtocol;
-    }
-
-    this.desiredProtocol = PROTOCOLS.MJSONWP;
-    const coercedVersion = semver.coerce(this.driverVersion);
-    if (!coercedVersion || coercedVersion.major < MIN_CD_VERSION_WITH_W3C_SUPPORT) {
+    if (_.includes(this._onlineStatus?.message, 'OperaDriver')) {
       this.log.info(
-        `The ChromeDriver v. ${this.driverVersion} does not fully support ${PROTOCOLS.W3C} protocol. ` +
-          `Defaulting to ${PROTOCOLS.MJSONWP}`,
+        `OperaDriver is known to not support ${PROTOCOLS.W3C} protocol, ` +
+        `defaulting to ${PROTOCOLS.MJSONWP}. ` +
+        `See https://github.com/operasoftware/operachromiumdriver/issues/88`
       );
-      return this.desiredProtocol;
+      this._desiredProtocol = PROTOCOLS.MJSONWP;
+      return this._desiredProtocol;
```

Comment:
Opera driver can speak W3C protocol technically https://gist.github.com/KazuCocoa/47614cc2d312452310f261b746281b9a

So, if we want to add operadriver specific handling, adding  `chromeOptions.w3c === true` and then `this._desiredProtocol = PROTOCOLS.W3C` is also needed

Reply:
updated